fhir-bucket/src/main/java/com/ibm/fhir/bucket/cos/CosClient.java

```diff
@@ -175,4 +184,40 @@ public void scan(String bucketName, String pathPrefix, Function<String, FileType
             }
         } while (running && result != null && result.isTruncated());
     }
+
+    /**
+     * Write the payload to the given bundleName as key
+     * @param bundleName
+     * @param payload
+     */
+    public void write(String bucketName, String objectName, String payload) {
+        byte[] raw = payload.getBytes(StandardCharsets.UTF_8);
+        
+        ObjectMetadata omd = new ObjectMetadata();
+        omd.setContentLength(raw.length);
+        omd.setContentEncoding("application/json");
+        
+        logger.info("Writing to COS '" + bucketName + ":" + objectName + "', bytes: " + raw.length);
+
+        if (DEBUG) {
+            try (FileOutputStream fos = new FileOutputStream(objectName)) {
+                fos.write(raw);
+                fos.flush();
+            } catch (IOException x) {
+                logger.log(Level.SEVERE, "Writing " + objectName, x);
+            }
+        } else {
+            // Write the object to the target key (objectName) in the given bucket
+            InputStream inputStream = new ByteArrayInputStream(raw);
+            PutObjectResult result = client.putObject(new PutObjectRequest(bucketName, objectName, inputStream, omd));
+            if (result != null) {
+                if (logger.isLoggable(Level.FINE)) {
+                    logger.fine("Wrote [" + bucketName + "]/" + objectName + ", ETag: " + result.getETag());
+                }
+            } else {
+                logger.warning("Writing failed for [" + bucketName + "]/" + objectName + ", bytes: " + raw.length);
```

Comment:
I assume the logformatter is injecting the thread ID so you can identifying which statement belongs to which thread?

Reply:
Just makes things more readable.

Reply:
That's fine - I was confirming the logformat enabled disambiguation.